src/brain/operating_unit_recorder.cpp

```diff
@@ -0,0 +1,382 @@
+#include <utility>
+
+#include "brain/operating_unit.h"
```

Comment:
Are we all decided on `brain` as the directory name for this stuff?

Reply:
No, we haven't decided yet. I think we can probably devise a better name after we have more stuff and the functionalities of the new components become more clear.